File: Array/0442.Find All Duplicates in an Array/Solution.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            while (nums[i] - 1 != i) {
                if (nums[i] == nums[nums[i] - 1]) {
                    break;
                }
                int tmp = nums[nums[i] - 1];
                nums[nums[i] - 1] = nums[i];
                nums[i] = tmp;
            }
        }
        List<Integer> ans = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            if (nums[i] - 1 != i) {
                ans.add(nums[i]);
            }
        }
        return ans;
    }
}

File: Array/0442.Find All Duplicates in an Array/Solution.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        for i in range(len(nums)):
            while nums[i] - 1 != i:
                if nums[i] == nums[nums[i] - 1]:
                    break
                nums[nums[i] - 1], nums[i] = nums[i], nums[nums[i] - 1]
        ans = []
        for i, v in enumerate(nums):
            if v - 1 != i:
                ans.append(v)
        return ans